src/HexagonOffload.cpp

```diff
@@ -909,9 +922,31 @@ Buffer<uint8_t> compile_module_to_hexagon_shared_object(const Module &device_cod
     auto bss = obj->find_section(".bss");
     if (bss != obj->sections_end()) {
         bss->set_alignment(128);
+        // TODO: We should set the type to SHT_NOBITS
+        // This will cause a difference in MemSize and FileSize like so:
+        //        FileSize = (MemSize - size_of_bss)
+        // When the Hexagon loader is used on 8998 and later targets,
+        // the difference is filled with zeroes thereby initializing the .bss
+        // section.
         bss->set_type(Elf::Section::SHT_PROGBITS);
+        debug(4) << "bss size = " << bss->get_size() << "\n";
+        Elf::Section::contents_iterator s = bss->contents_begin();
+        Elf::Section::contents_iterator e = bss->contents_end();
+        std::fill(s, e, 0);
+    }
+
+    auto dtors = obj->find_section(".dtors");
+    if (dtors != obj->sections_end()) {
+        dtors->append_contents((uint32_t) 0);
     }
 
+    auto ctors = obj->find_section(".ctors");
+    if (ctors != obj->sections_end()) {
+        ctors->prepend_contents((uint32_t) 0);
```

Comment:
Why prepend? (comment)

Reply:
I meant comment the code explaining what's going on here. Mainly the append/prepend logic, and also thinking about it more now, how does this data get referenced? I didn't see a relocation referring to this?

Reply:
Oh dang it sorry, I misunderstood  ;)